runtime/common/src/auctions.rs

```diff
@@ -754,92 +755,42 @@ mod tests {
 		type MaxFreezes = ConstU32<1>;
 	}
 
-	#[derive(Eq, PartialEq, Ord, PartialOrd, Clone, Copy, Debug)]
-	pub struct LeaseData {
-		leaser: u64,
-		amount: u64,
-	}
-
-	thread_local! {
-		pub static LEASES:
-			RefCell<BTreeMap<(ParaId, BlockNumber), LeaseData>> = RefCell::new(BTreeMap::new());
-	}
-
-	fn leases() -> Vec<((ParaId, BlockNumber), LeaseData)> {
-		LEASES.with(|p| (&*p.borrow()).clone().into_iter().collect::<Vec<_>>())
-	}
-
-	pub struct TestLeaser;
-	impl Leaser<BlockNumber> for TestLeaser {
-		type AccountId = u64;
-		type LeasePeriod = BlockNumber;
-		type Currency = Balances;
-
-		fn lease_out(
-			para: ParaId,
-			leaser: &Self::AccountId,
-			amount: <Self::Currency as Currency<Self::AccountId>>::Balance,
-			period_begin: Self::LeasePeriod,
-			period_count: Self::LeasePeriod,
-		) -> Result<(), LeaseError> {
-			LEASES.with(|l| {
-				let mut leases = l.borrow_mut();
-				let now = System::block_number();
-				let (current_lease_period, _) =
-					Self::lease_period_index(now).ok_or(LeaseError::NoLeasePeriod)?;
-				if period_begin < current_lease_period {
-					return Err(LeaseError::AlreadyEnded)
-				}
-				for period in period_begin..(period_begin + period_count) {
-					if leases.contains_key(&(para, period)) {
-						return Err(LeaseError::AlreadyLeased)
-					}
-					leases.insert((para, period), LeaseData { leaser: leaser.clone(), amount });
-				}
-				Ok(())
-			})
-		}
+	// This is procedural does not exists already,
+	// but it's like we would want to build mock pallets.
+	#[mock_builder::pallet]
+	mod pallet_mock_leaser {
+		trait Leaser<BlockNumber> {
+			type AccountId;
+			type LeasePeriod;
+			type Currency: ReservableCurrency<Self::AccountId>;
```

Comment:
This can be placed here or under a *mocks* folder where they can be reused. Since the generics of the trait can be specified through the `Config` trait, it can be easily shared.